SPHINXsys/src/shared/io_system/in_output.h

```diff
@@ -230,6 +230,27 @@ namespace SPH {
 		VtuStringData _vtuData;
 	};
 
+	/**
+	 * @class BodyStatesRecordingToVtuStringRunTime
+	 * @brief  Writes simulation results as strings for bodies in rum time
+	 * the output is a std::pair of strings with VTK XML format that can be visualized by ParaView
+	 * the data type vtkUnstructedGrid
+	 */
+	class BodyStatesRecordingToVtuStringRunTime : public BodyStatesRecordingToVtu
+	{
+	public:
+		BodyStatesRecordingToVtuStringRunTime(In_Output& in_output, SPHBodyVector bodies);
+		virtual ~BodyStatesRecordingToVtuStringRunTime() = default;
+
+		using VtuStringDataRunTime = std::pair<std::string, std::string>;
+
+		const VtuStringDataRunTime& GetVtuDataRunTime() const;
+	protected:
+		virtual void writeWithFileName(const std::string& sequence) override;
```

Comment:
I think you dont need virtual keyword since this function is already override the original virtual function